src/Raven.Server/RavenServer.cs

```diff
@@ -486,6 +542,87 @@ private void StartMonitoringCpuCredits()
             }
         }
 
+        public class CpuCreditsResponse
+        {
+            public double Remaining { get; set; }
+            public DateTime Timestamp { get; set; }
+        }
+
+        private CpuCreditsResponse GetCpuCreditsFromExec()
+        {
+            var command = Configuration.Server.CpuCreditsExec;
+            var arguments = Configuration.Server.CpuCreditsExecArguments;
+
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = command,
+                Arguments = arguments,
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                CreateNoWindow = true
+            };
+
+            Process process;
+
+            try
+            {
+                process = Process.Start(startInfo);
+            }
+            catch (Exception e)
+            {
+                throw new InvalidOperationException($"Unable to get cpu credits by executing {command} {arguments}. Failed to start process.", e);
+            }
+
+            using (var ms = new MemoryStream())
```

Comment:
This is a lot of duplicated code.
We know that we want to change this already, so let's centralized process execution

Reply:
Will open an issue for that